db.go

```diff
@@ -349,22 +377,31 @@ type DB struct {
 	IgnoreUnmappedCols bool
 	Context            context.Context
 	Logger             QueryLogger
-	mu                 sync.RWMutex
-	models             map[reflect.Type]*Model
-	mappings           map[reflect.Type]fieldMap
+	// Query deadline, ignored if Deadline.IsZero().
+	Deadline time.Time
+	// Function to add a deadline to a query.
+	DeadlineQueryRewriter func(db *DB, query interface{}, millis int64) (queryWithDeadline interface{}, err error)
```

Comment:
This would read better if you extracted return type into its own named type. It would also let you attach a comment to the name type that can guide implementers and help them implement this correctly.